libcaf_core/caf/inspector_access_type.hpp

```diff
@@ -80,98 +65,23 @@ struct inspector_access_type {
 
 /// @relates inspector_access_type
 template <class Inspector, class T>
-constexpr auto inspect_object_access_type() {
+constexpr auto inspect_access_type() {
   using namespace detail;
-  // Order: unsafe (error) > C Array > builtin_inspect > inspector_access >
-  // inspect > trivial > defaults.
+  // Order: unsafe (error) > C Array > builtin > builtin_inspect
+  //        > inspector_access > inspect > trivial > defaults.
   if constexpr (is_allowed_unsafe_message_type_v<T>)
     return inspector_access_type::unsafe{};
   else if constexpr (std::is_array<T>::value)
-    return inspector_access_type::array{};
-  else if constexpr (has_builtin_inspect<Inspector, T>::value)
-    return inspector_access_type::builtin{};
-  else if constexpr (detail::is_complete<inspector_access<T>>)
-    return inspector_access_type::specialization{};
-  else if constexpr (has_inspect_overload<Inspector, T>::value)
-    return inspector_access_type::inspect{};
-  else if constexpr (is_trivial_inspector_value_v<Inspector::is_loading, T>)
-    return inspector_access_type::trivial{};
-  else if constexpr (std::is_integral<T>::value)
-    return inspector_access_type::integral{};
-  else if constexpr (std::is_enum<T>::value)
-    return inspector_access_type::enumeration{};
-  else if constexpr (std::is_empty<T>::value)
-    return inspector_access_type::empty{};
-  else if constexpr (is_stl_tuple_type_v<T>)
     return inspector_access_type::tuple{};
```

Comment:
Lists are dynamically sized. An array ist just a special case of tuples where each element has the same type. Not just my opinion, since the standard specializes `std::get`, `std::tuple_element`, etc. for `std::array` for exactly this reason. 🙂